lib/views/settings_view/custom_playback_speed_view.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+
+void showAddCustomSpeedDialog(
+  BuildContext context,
+  Function(double) onSpeedAdded,
+) {
+  double customSpeed = 1.0;
+  String errorMessage = '';
+
+  void validateAndSetSpeed(String value) {
+    if (value.isEmpty) {
+      errorMessage = '';
+    } else {
+      double? parsedValue = double.tryParse(value);
+      if (parsedValue != null && parsedValue >= 0.25 && parsedValue <= 4.0) {
+        customSpeed = parsedValue;
+        errorMessage = '';
```

Comment:
Maybe we can add this to check length

```suggestion
        errorMessage = '';
        if(parsedValue.toString().length > 4) {
          errorMessage = 'Number is too long';
        }
```

Reply:
thanks achraf, I just implemented your suggestion but to be honest I'm not sure if this is necessary or not. because       

```
if (parsedValue != null && parsedValue >= 0.25 && parsedValue <= 4.0) {
        customSpeed = parsedValue;
        errorMessage = '';


```
this part is already check the input is between 0.25-4 range, and if it s not it gonna show an error message. I already check if the number is too long or not. So, as your suggestion it looks like this. Are you sure it is necessary to add this change? :

<img width="290" alt="Screenshot 2024-01-11 at 13 28 47" src="https://github.com/TUM-Dev/gocast_mobile/assets/43902289/51d6a500-76bf-4a25-b1ac-755da0693920">
<img width="290" alt="Screenshot 2024-01-11 at 13 28 51" src="https://github.com/TUM-Dev/gocast_mobile/assets/43902289/850941d3-2da1-4bcf-bf54-eea27df6a1ea">

Reply:
Yes but numbers like 0.2643242347325472 shouldn´t be allowed

Reply:
With my suggestions we limit it to 2 numbers after the zero :)

Reply:
implemented in the last commit